tools/aliaslookup/aliaslookup.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+"""Utility to look up OSV records by alias.
+"""
+
+from google.cloud import datastore
+
+import sys
+import fileinput
+import argparse
+
+
+def LookupByAliases(client: datastore.Client,
+                    identifiers: list[str],
+                    verbose=False) -> datastore.query.Iterator:
+  """Look up OSV records by alias.
+
+  Args:
+    client: a datastore.Client object.
+    identifiers: a list of strings being the aliases to look up.
+    verbose: a boolean whether to emit more verbose processing information.
+
+  Returns:
+    a datastore.query.Iterator
+  """
+  query = client.query(kind="Bug")
+  # the "IN" operator only supports 30 identifiers as a maximum, so we need to
+  # batch into multiple a composite OR query
+  # This still caps out at 100 total filters, sadly
+  query.add_filter(
+      filter=datastore.query.PropertyFilter("aliases", "IN", identifiers))
+  if verbose:
+    print(f"Running query {query.filters[0]} "
+          f"on {query.kind} (in {query.project})...")
+  result = list(query.fetch())
+  if verbose:
+    print(f"Retrieved {len(result)} bugs")
+
+  return result
+
+
+def main() -> None:
+  parser = argparse.ArgumentParser(description="Look up OSV records by alias")
+  parser.add_argument(
+      "--aliases",
+      action="store",
+      dest="aliases",
+      help="comma-separated list of IDs to look up")
+  parser.add_argument(
+      "--filename",
+      action="store",
+      dest="filename",
+      help="Filename of newline-separated IDs to look up")
+  parser.add_argument(
+      "--project",
+      action="store",
+      dest="project",
+      default="oss-vdb-test",
+      help="GCP project to operate on")
+  parser.add_argument(
+      "--verbose",
+      action=argparse.BooleanOptionalAction,
+      dest="verbose",
+      default=False,
+      help="Be more verbose")
+
+  args = parser.parse_args()
+
+  client = datastore.Client(project=args.project)
+
+  aliases = []
+  bugs = []
+
+  if not args.aliases and not args.filename:
+    if sys.stdin.isatty():
+      print("Reading newline separated IDs from STDIN...")
+    for alias in sys.stdin:
+      aliases.append(alias.strip())
+
+  if args.aliases:
+    aliases = args.aliases.split(",")
+
+  if args.filename:
+    with fileinput.input(files=args.filename) as f:
+      for alias in f:
+        aliases.append(alias.strip())
+
+  if aliases:
+    bugs = LookupByAliases(client, aliases, args.verbose)
+
+  if bugs:
+    print("aliases,bug")
+    # TODO: figure out if pagination needs to be handled here...
+    for bug in bugs:
+      print(f"{bug['aliases'][0]},{bug['db_id']}")
```

Comment:
Should this be printing all the aliases?

Reply:
Ah, good catch, it's actually more nuanced, just the matching aliases. Fixed in https://github.com/google/osv.dev/pull/1198/commits/e64c34e9b8fda220d1084304bd8567f1d1486865